doc/api/assert.md

```diff
@@ -147,6 +147,136 @@ try {
 }
 ```
 
+## Class: `assert.CallTracker`
+
+### `new assert.CallTracker()`
+<!-- YAML
+added: REPLACEME
+-->
+
+Creates a new [`CallTracker`][] object which can be used to track if functions
+were called a specific number of times. The `tracker.verify()` must be called
+for the verification to take place. The usual pattern would be to call it in a
+[`process.on('exit')`][] handler.
+
+```js
+const assert = require('assert');
+
+const tracker = new assert.CallTracker();
+
+function func() {}
+
+// callsfunc() must be called exactly 1 time before tracker.verify().
+const callsfunc = tracker.calls(func, 1);
+
+callsfunc();
+
+// Calls tracker.verify() and verifies if all tracker.calls() functions have
+// been called exact times.
+process.on('exit', () => {
+  tracker.verify();
+});
+```
+
+### `tracker.calls([fn][, exact])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fn` {Function} **Default** A no-op function.
+* `exact` {number} **Default** `1`.
+* Returns: {Function} that wraps `fn`.
+
+The wrapper function is expected to be called exactly `exact` times. If the
+function is not called exactly `exact` times before
+[`tracker.verify()`][] is called, an exception is thrown.
```

Comment:
Non-blocking nit: Although it appears elsewhere in our docs, I've always disliked the formulation "an exception is thrown." The passive voice makes it ambiguous. By switching to active, it clarifies *what* throws the exception, which can be important. (It's an even bigger issue with the even more pervasive "is emitted" for events throughout our docs.)

All that said, finding clear active-voice wording that doesn't have its own awkwardness can be challenging, but maybe this?:

```suggestion
[`tracker.verify()`][] is called, then `tracker.verify()` will throw an exception.
```

Reply:
Aside: We have both `an exception is thrown` and `an error is thrown` throughout our docs. We should probably standardize on one or the other. I think _error_ is more JavaScript-y than _exception_ but I mostly care about consistency on this. @nodejs/documentation 

```suggestion
[`tracker.verify()`][] is called, then [`tracker.verify()`][] will throw an error.
```

Reply:
@trott `error` is an object that is an instance of a subclass of `Error`, `exception` is a value that is passed to `throw` (which can be any JS value). “Throwing an exception” is never wrong, “Throwing an error” is more specific but generally accurate.